src/main/java/org/fluentd/logger/sender/AsyncRawSocketSender.java

```diff
@@ -24,7 +24,7 @@
  * @author mxk
  *
  */
-public class AsyncRawSocketSender implements Sender {
+public class AsyncRawSocketSender implements AsyncSender {
```

Comment:
Ah, this class should implements `Sender` because `FluentLogger` can have only `Sender` instance not `AsyncSender`. https://github.com/fluent/fluent-logger-java/blob/master/src/main/java/org/fluentd/logger/FluentLogger.java#L65

Before the latest commits, I think only one issue was https://github.com/fluent/fluent-logger-java/pull/49#discussion_r62480706. So if we compromise on it, it'll be good enough as an experimental feature.

How about always returning true from `AsyncRawSocketSender#emit` to keep its asynchronous? It sounds reasonable compromise to me.